awx/main/tests/functional/api/test_workflow_node.py

```diff
@@ -132,11 +135,41 @@ def test_activity_stream_create_wf_approval(self, post, admin_user, workflow_job
 
         qs2 = ActivityStream.objects.filter(organization__isnull=True)
         assert qs2.count() == 5
-        assert qs2[0].operation == 'create'
-        assert qs2[1].operation == 'create'
-        assert qs2[2].operation == 'create'
-        assert qs2[3].operation == 'create'
-        assert qs2[4].operation == 'update'
+        assert list(qs2.values_list('operation', 'object1')) == [('create', 'user'),
+                                                                 ('create', 'workflow_job_template'),
+                                                                 ('create', 'workflow_job_template_node'),
+                                                                 ('create', 'workflow_approval_template'),
+                                                                 ('update', 'workflow_job_template_node'),
+                                                                 ]
+
+    @pytest.mark.django_db
+    def test_approval_node_actions(self, post, admin_user, job_template):
+        # This test ensures that a user (with permissions to do so) can approve/deny
+        # workflow approvals.  Also asserts that trying to approve/deny approvals
+        # that have already been dealt with will throw an error.
+        wfjt = WorkflowJobTemplate.objects.create(name='foobar')
+        node = wfjt.workflow_nodes.create(unified_job_template=job_template)
+        url = reverse('api:workflow_job_template_node_create_approval',
+                      kwargs={'pk': node.pk, 'version': 'v2'})
+        post(url, {'name': 'Approve/Deny Test', 'description': '', 'timeout': 0},
+             user=admin_user, expect=200)
+        post(reverse('api:workflow_job_template_launch', kwargs={'pk': wfjt.pk}),
+             user=admin_user, expect=201)
+        wf_job = WorkflowJob.objects.first()
+        TaskManager().schedule()
```

Comment:
Are these `.schedule()` calls necessary?

Reply:
It's certainly possible to emulate the behavior, but it's more than just changing the WFJT status to pending. We also need to spawn the approval job.

It's true those could both be emulated, but I would also prefer to have some test (no matter which) hit the task manager logic with approval jobs in the mix. That gives regression coverage we do not otherwise have in these tests, because this is a job type we did not have before.

Reply:
Okay, I guess that's fair 👍